asv/benchmarks.py

```diff
@@ -462,26 +468,67 @@ def run_benchmarks(self, env, show_stderr=False, quick=False, profile=False,
 
             - `result`: The numeric value of the benchmark (usually
               the runtime in seconds for a timing benchmark), but may
-              be an arbitrary JSON data structure. For parameterized tests, 
+              be an arbitrary JSON data structure. For parameterized tests,
               this is a dictionary with keys 'params' and 'result', where
               the value of 'params' contains a list of lists of parameter values,
               and 'result' is a list of results, corresponding to itertools.product
-              iteration over parameters. 
+              iteration over parameters.
               Set to `None` if the benchmark failed.
 
             - `profile`: If `profile` is `True`, this key will exist,
               and be a byte string containing the cProfile data.
         """
         log.info("Benchmarking {0}".format(env.name))
-        with log.indent():
-            times = {}
-            benchmarks = sorted(list(six.iteritems(self)))
-            for name, benchmark in benchmarks:
-                if skip and name in skip:
-                    continue
-                times[name] = run_benchmark(
-                    benchmark, self._benchmark_dir, env, show_stderr=show_stderr,
-                    quick=quick, profile=profile)
+
+        cache_file = tempfile.NamedTemporaryFile(suffix='dbm', delete=False)
+        cache_file.close()
+        # We have to actually delete the file here, since the database
+        # type can not be determined from a zero-length file.  We
+        # can't create the database in the master process, since the
+        # file formats available may vary between Python environments.
+        os.remove(cache_file.name)
```

Comment:
I think this allows a local race condition attack with arbitrary code execution.
`mkdtemp` is probably a cleaner solution without such problems.